### PR TITLE
[dv] Fix up prototype of dv_base_reg_block::build

### PR DESCRIPTION
**Pull request by rswarbrick**
This gained a "csr_excl" parameter in commit 85aca77, but we forgot to
update all deriving classes to have the same prototype. VCS seems to
allow us to be slapdash like this, but Xcelium shouted. We "fixed"
that by giving the base class implementation a default argument in
commit 1b1f2ec. This shut Xcelium up, but seems not to actually follow
the IEEE 1800-2017 standard (see section 8.20 "Virtual methods").

This patch removes the default value from csr_excl (since it didn't
help anyway, we may as well remove the magic) and fixes up the
deriving classes.

See discussion in issue #3300.

### CODE REVIEW

File: hw/ip/nmi_gen/dv/env/nmi_gen_reg_block.sv [hunk 1]
```diff
@@ -227,7 +227,7 @@ class nmi_gen_reg_block extends dv_base_reg_block;
     super.new(name, has_coverage);
   endfunction : new
 
-  virtual function void build(uvm_reg_addr_t base_addr);
+  virtual function void build(uvm_reg_addr_t base_addr, csr_excl_item csr_excl);
```
Comment (sriyerg) [inline hw/ip/nmi_gen/dv/env/nmi_gen_reg_block.sv:230]: These files should be removed, given that we now rely on the fusesoc generator to create this when running the sims.

File: hw/dv/sv/dv_base_reg/dv_base_reg_block.sv [hunk 2]
```diff
@@ -13,8 +13,7 @@ class dv_base_reg_block extends uvm_reg_block;
   endfunction
 
   // provide build function to supply base addr
-  virtual function void build(uvm_reg_addr_t base_addr,
-                              csr_excl_item csr_excl = null);
+  virtual function void build(uvm_reg_addr_t base_addr, csr_excl_item csr_excl);
```
Comment (sriyerg) [inline hw/dv/sv/dv_base_reg/dv_base_reg_block.sv:16]: Please revert this change.

**Comment by sriyerg** [comment]
Its not the default value to the csr_excl object arg that is causing the problem - it's the stale files for these IP level testbenches, which should really be removed.

**Comment by rswarbrick** [comment]
Ok, it looks like these files are just stale and should have been deleted when we changed dvsim to generate them on the fly. Does that apply to both `nmi_reg_block.sv` and `rv_dm_reg_block.sv`? If so, it looks like we should just close this PR and open one that deletes those files, right?

**Comment by sriyerg** [comment]
> Ok, it looks like these files are just stale and should have been deleted when we changed dvsim to generate them on the fly. Does that apply to both `nmi_reg_block.sv` and `rv_dm_reg_block.sv`? If so, it looks like we should just close this PR and open one that deletes those files, right?

Yes. both can be deleted. The ral generation section needs to be added to the env core file.
I am not sure why `nmi_gen` even has a testbench.

**Comment by rswarbrick** [comment]
I'm going to push up a patch deleting the `nmi_gen` reg block (it doesn't have a testbench otherwise). The debug module stuff... is completely broken. The core files are missing entries, and there are loads of bits that just don't work. I reckon we just leave this bit-rotted until someone picks it up properly.
